File: ToDo/ocaml/4 - OCaml Piscine - ready/me125/ocaml_pool-master/d00/ex02/ft_power.ml
(* ************************************************************************** *)
(*                                                                            *)
(*                                                        :::      ::::::::   *)
(*   ft_power.ml                                        :+:      :+:    :+:   *)
(*                                                    +:+ +:+         +:+     *)
(*   By: jaguillo <jaguillo@student.42.fr>          +#+  +:+       +#+        *)
(*                                                +#+#+#+#+#+   +#+           *)
(*   Created: 2015/06/15 13:44:48 by jaguillo          #+#    #+#             *)
(*   Updated: 2015/06/15 13:58:10 by jaguillo         ###   ########.fr       *)
(*                                                                            *)
(* ************************************************************************** *)

let rec ft_power n pow =
	if pow <= 0 then
		1
	else
		n * (ft_power n (pow - 1))

(*
** Test
*)
let test a b =
	print_string "Test with ["; print_int(a);
	print_string ", "; print_int(b); print_string "]: ";
	print_int (ft_power a b); print_char '\n'

let () = test 2 4;
	test 3 0;
	test 0 5;
	test (-2) 5;
	test 2 8

File: ToDo/pushswap/2 - Push_Swap - ready/8!!!!!!!!!!!!!!!!!/push_swap-master/libft/srcs/ft_power.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   ft_power.c                                         :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: ljoly <ljoly@student.42.fr>                +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2017/02/24 16:20:02 by ljoly             #+#    #+#             */
/*   Updated: 2017/02/24 16:20:03 by ljoly            ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include "libft.h"

long long int		ft_power(int nb, int power)
{
	long long int	r;

	if (power == 0)
		return (1);
	if (power < 0)
		return (0);
	r = nb;
	while (power > 1)
	{
		r = r * nb;
		power--;
	}
	return (r);
}
